### PR TITLE
[NEXUS-8855] resolve relative source against remote URL when checking redirects

### PR DESCRIPTION
**Pull request by mcculls**
and improve logging as requested in https://issues.sonatype.org/browse/NEXUS-8855

http://bamboo.s/browse/NX-OSSF1094-2 :white_check_mark:

### CODE REVIEW

**Comment by adreghiciu** [comment]
+1

**Comment by peterlynch** [comment]
+1

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientManagerImpl.java [hunk 1]
```diff
@@ -230,24 +230,40 @@ public boolean isRedirected(final HttpRequest request, final HttpResponse respon
                 " from proxy " + proxyRepository + " but no location present");
           }
 
+          final String repoId = proxyRepository.getId();
+
           // Some complication here as it appears that something may be null, but its not clear what was null
```
Comment (mrprescott) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientManagerImpl.java]: This comment is hilarious. (I know it predates this PR.)

**Comment by mrprescott** [comment]
+1

**Comment by mcculls** [comment]
@peterlynch is 4ac5ee5 what you were thinking of?

**Comment by peterlynch** [comment]
@mcculls much better :+1:

**Comment by mcculls** [comment]
Merged to nexus-2.11.x as 119b2f4

**Pull request closed without merging**